spec/models/spree/payment_spec.rb

```diff
@@ -504,11 +513,40 @@
           end
         end
 
+        context "when amount <= credit_allowed" do
+          it "makes the state processing" do
+            payment.payment_method.name = 'Gateway'
+            payment.payment_method.distributors << create(:distributor_enterprise)
+            payment.payment_method.save!
+
+            payment.order = create(:order)
+
+            payment.state = 'completed'
+            allow(payment).to receive(:credit_allowed) { 10 }
+            payment.partial_credit(10)
+            expect(payment).to be_processing
+          end
+
+          it "calls credit on the source with the payment and amount" do
+            payment.state = 'completed'
+            allow(payment).to receive(:credit_allowed) { 10 }
+            expect(payment).to receive(:credit!).with(10)
+            payment.partial_credit(10)
+          end
```

Comment:
These two specs are in the same `context`, but seem to duplicate the setup. can it be moved to `let`/`before` blocks?
The second example doesn't seem to need as much setup as the first, so I wonder if it's all necessary.